### PR TITLE
[Core][CompareTwoFilesCheckProcess] updating deprecated calls

### PR DESCRIPTION
**Pull request by philbucher**
**Description**
after #8612

### CODE REVIEW

File: kratos/python_scripts/compare_two_files_check_process.py [hunk 1]
```diff
@@ -82,9 +82,9 @@ def ExecuteFinalize(self):
         Please see the respective files for details on the format of the files
         """
 
-        KratosMultiphysics.DataCommunicator.GetDefault().Barrier()
+        KratosMultiphysics.Testing.GetDefaultDataCommunicator().Barrier()
```
Comment (rubenzorrilla) [inline kratos/python_scripts/compare_two_files_check_process.py:85]: Just for me to know. Shouldn't we take it from the parallel environment in here? (even though this is normally used for the testing we are out of the scope of a test).

**Review by rubenzorrilla — APPROVED** [review]

File: kratos/python_scripts/compare_two_files_check_process.py (hunk 1 above)
Comment (philbucher) [reply kratos/python_scripts/compare_two_files_check_process.py:85]: Usually I would agree with you, but this class derives from Unittest, hence I went for the testing one:
https://github.com/KratosMultiphysics/Kratos/blob/d200e8fd4fdb2c5fead72a04538cfc2d560537fd/kratos/python_scripts/compare_two_files_check_process.py#L19

Comment (rubenzorrilla) [reply kratos/python_scripts/compare_two_files_check_process.py:85]: Makes sense for me.

Comment (philbucher) [reply kratos/python_scripts/compare_two_files_check_process.py:85]: probably we should move this file to the `testing` subfolder

**Pull request merged**